src/entity/Entity.php

```diff
@@ -1535,6 +1543,25 @@ public function despawnFromAll() : void{
 		}
 	}
 
+	/**
+	 * Returns the item that can be used to spawn this entity, by default it returns null if there isn't one
+	 */
+	public function getSpawnItem() : ?Item{
+		return VanillaItems::AIR();
+	}
+
+	/**
+	 * Returns the item that players will equip when middle-clicking on this entity.
+	 * If addUserData is true, additional data may be added, such as name tag, fire ticks, etc.
+	 */
+	public function getPickedItem() : ?Item{
+		$item = $this->getSpawnItem();
+		if($item === null){
+			return null;
+		}
+		return $item;
```

Comment:
This can be simplified to just return `getSpawnItem()` directly. Perhaps the two separate functions are redundant.